content/blog/2025-03-27-valkey-8-1-0-rc1.md

```diff
@@ -0,0 +1,282 @@
+
+### I/O Threads Improvements
+
+Following up the I/O threads improvements added in 8.0, more operations have been offloaded to the I/O thread pool in the 8.1 release, improving the throughput and latency of some operations.
+
+In the new release, TLS connections are now able to offload the TLS negotiation to I/O threads. This change improves the rate of accepting new connections by around [300%](https://github.com/valkey-io/valkey/pull/1338).
+
+Other sources of overhead in the TLS connection handling were identified, namely in the calls to `SSL_pending()` and `ERR_clear_error()` functions, which were being called in the main event thread. By offloading these functions to the I/O threads pool, a throughput improvement was achieved in some operations. For instance, it was observed a [10%](https://github.com/valkey-io/valkey/pull/1271) improvement in `SET` operations throughput, and a [22%](https://github.com/valkey-io/valkey/pull/1271) improvement in `GET` operations throughput.
+
+Replication traffic efficiency was also improved in 8.1 by offloading the reading of replication stream on the replicas to the I/O thread pool which means they can serve more read traffic. On the primaries, replication stream writes are now offloaded to the I/O thread pool.
+
+### Replication Improvements
+
+Full syncs with TLS enabled benefit from an [18%](https://github.com/valkey-io/valkey/pull/1479 ) improvement by removing CRC checksummming.
+
+The fork copy-on-write memory overhead is reduced by up to [47%](https://github.com/valkey-io/valkey/pull/905)
+ 
+### Sorted set and hyperloglog and bitcount optimisations
+
+`ZRANK` command, which serves a popular usecase in operating Leaderboards, was optimised to perform up to [45%](https://github.com/valkey-io/valkey/pull/1389) faster, depending on the serted set size.
+
+The probabilistic hyperloglog is another great data type, used for counting unique elements in very large datasets whilst using only 12KB of memory regardless of the amount of elements. By using the modern CPUs Advanced Vector Extensions of x86, Valkey 8.1 can achieve a [12x](https://github.com/valkey-io/valkey/pull/1293) speed for the operations like `PFMERGE` and `PFCOUNT` on hyperloglog data types.
+
+Similarly, the BITCOUNT operation has been improved up to [514%](https://github.com/valkey-io/valkey/pull/1741) using AVX2 on x86.
+
+### Other Performance Enhancements
+
+Active Defrag has been improved to eliminate large latencies to below [1ms](https://github.com/valkey-io/valkey/pull/1242) and anti-starvation protection has been introduced in the presence of long-running commands.
+
+Periodic processing/monitoring of clients has been moved off of the 100ms `serverCron()` interval loop, reducing system [overhead](https://github.com/valkey-io/valkey/pull/1387)
+ when large numbers of clients are present. 
+
+## Observability
+
+There are also several improvements to the observability of the system behavior in Valkey 8.1.
```

Comment:
Maybe expand a bit in the observability area to name a few new metrics added and a short description why they matter?